mypy/test/data.py

```diff
@@ -277,7 +277,7 @@ class DataDrivenTestCase(pytest.Item):
 
     def __init__(
         self,
-        parent: DataSuiteCollector,
+        parent: DataFileCollector,
```

Comment:
Could you add an assert to `__init__` to make sure it's only `DataFileCollector`?

Reply:
Sure. Done!